examples/Azure_IoT_Central_ESP32_AzureIoTKit/AzureIoT.cpp

```diff
@@ -94,7 +94,15 @@ void azure_iot_init(azure_iot_t* azure_iot, azure_iot_config_t* azure_iot_config
     _az_PRECONDITION(is_az_span_empty(azure_iot_config->dps_id_scope));
     _az_PRECONDITION(is_az_span_empty(azure_iot_config->dps_registration_id));
   }
-  _az_PRECONDITION_VALID_SPAN(azure_iot_config->device_key, 1, false);
+
+  // Either device key or device certificate and certificate key should be defined.
+  if (is_az_span_empty(azure_iot_config->device_key) &&
+    (is_az_span_empty(azure_iot_config->device_certificate) || is_az_span_empty(azure_iot_config->device_certificate_private_key) ))
+  {
+    LogError("Please define either a device key or a device certificate and certificate private key. See iot_configs.h");
+    _az_PRECONDITION(false);
```

Comment:
Same as above - this is either a complete precondition or just a runtime error.

Reply:
hmmmmm, that's tricky. There is no way to have a compound boolean check with preconditions. That's because **either** device_key is NULL and cert&private key are NOT, or device_key is NOT NULL and cert&private key are NULL.
Not sure that's possible to check only with preconditions.

Reply:
Btw @danewalton , the code above does not fail if both device key and certs are provided.

Reply:
I have it just logging and returning. The #define structure shouldn't let this happen and things will blow up on connection if this is hit. The log message will still be spit out to tell people how to fix it.